test/integration/driver.go

```diff
@@ -232,28 +231,34 @@ func getRestartEpoch(pod string) (int, error) {
 	return 0, fmt.Errorf("could not obtain envoy restart epoch")
 }
 
-func addConfig(config []byte, kind, name string) {
+func addConfig(config []byte, kind, name string, create bool) {
 	log.Println("Add config")
 	log.Println(string(config))
-	out, err := yaml.YAMLToJSON(config)
-	check(err)
 	istioKind, ok := model.IstioConfig[kind]
 	if !ok {
 		check(fmt.Errorf("Invalid kind %s", kind))
 	}
-	v, err := istioKind.FromJSON(string(out))
+	v, err := istioKind.FromYAML(string(config))
 	check(err)
-	check(istioClient.Put(model.Key{
+	key := model.Key{
 		Kind:      kind,
 		Name:      name,
 		Namespace: params.namespace,
-	}, v))
+	}
+	if create {
+		check(istioClient.Post(key, v))
+	} else {
+		check(istioClient.Put(key, v))
+	}
 }
```

Comment:
I think with just one create command, it would be slightly confusing to the end user. Especially, given that kubernetes has `kubectl apply` (which provides update semantics). In the interest of clarity, it might be useful to explicitly have create, update, get and delete. What do you think?